compiler/modules/types/list.bal

```diff
@@ -572,13 +572,60 @@ function listSubtypeIntersect(Context cx, SubtypeData t1, SubtypeData t2) return
 }
 
 function listSubtypeDiff(Context cx, SubtypeData t1, SubtypeData t2) returns SubtypeData {
+    MemoBddCache cache = new (cx.listMemo);
+    if listIsEmptySimple(cx, bddIntersect(cache, <Bdd> t1, <Bdd> t2)) is true {
+        return t1;
+    }
     Bdd b = memoSubtypeDiff(cx.listMemo, <Bdd>t1, <Bdd>t2);
     if b != false && cx.listMemo[b] == () {
         //io:println("fresh BDD: ", bddToString(b));
     }
     return b;
 }
 
+function listIsEmptySimple(Context cx, Bdd bdd) returns boolean? {
+    BddMemo? m = cx.listMemo[bdd];
+    if m !is () && m.empty is boolean {
+        return <boolean>m.empty;
+    }
+    boolean empty = bddEvery(cx, bdd, (), (), listFormulaIsDefinitelyEmpty);
+    if empty == true {
+        cx.listMemo.put({ bdd, empty }); // handle both memoized values being () and not in memo
```

Comment:
More idiomatic: `cx.listMemo[bdd] = empty;`

Reply:
I don't think this is possible due `cannot update 'table<wso2/nballerina.types:0.1.0:BddMemo> key(bdd)' with member access expression` error

Reply:
My suggestion was wrong.